src/components/PopupTrigger.js

```diff
@@ -51,21 +59,32 @@ const PopupTrigger = Utils.forwardRefWithName('PopupTrigger', ({ content, childr
 
   const child = Children.only(children)
   const childId = child.props.id || id
+  const labelledby = child.props['aria-labelledby'] || childId
+
   return h(Fragment, [
     cloneElement(child, {
       id: childId,
       className: `${child.props.className || ''} ${childId}`,
       onClick: (...args) => {
         child.props.onClick && child.props.onClick(...args)
         setOpen(!open)
-      }
+      },
+      'aria-haspopup': role, // 'dialog', 'listbox', 'menu' are valid values
+      'aria-expanded': open,
+      'aria-controls': open ? menuId : undefined,
+      'aria-owns': open ? menuId : undefined
     }),
     open && h(Popup, {
+      id: menuId,
       target: childId,
       handleClickOutside: () => setOpen(false),
       outsideClickIgnoreClass: childId,
       onClick: closeOnClick ? () => setOpen(false) : undefined,
-      ...props
+      popupProps: {
+        role,
+        'aria-labelledby': labelledby,
+        ...props
```

Comment:
Passing props here as opposed to a level up is technically a functional change. I would be worried about props not relating to the popup that should be passed in getting lost. However, it looks like we weren't using the general props there anyway, so it might be fine? Just worth double checking.

Reply:
So the problem with just pulling in ...props in `Popup` was that it also grabs all of the internal `onClickOutside` properties for some reason. I couldn't just stick them straight into the `div`, so I did it this way instead to only pass in the ones we actually care about.

But now that I think about it, you're right! Properties `Popup` needs, like `side`, need to be passed in one level further up since they're not explicitly pulled out in `PopupTrigger`.

Can I just fix that by pulling those out explicitly in `PopupTrigger`?

Reply:
It looks like the only unnamed prop which could possibly be passed through was `side`, so I just pulled that out here. I decided not to push them into `popupProps` inside the `PopupTrigger`, because it could change existing calls, and then it's not clear what should be in versus out. For example, I'm pulling `role` out explicitly because I'm using it elsewhere, however it also gets attached to the DOM node so it should by that logic be in popupProps instead. Better to just leave it all in props and do the right thing. :)

Reply:
Nevermind! Users of the `PopupTrigger` can also pass in properties to be consumed by the `onClickOutside` functionality. I forgot about that. I couldn't find anywhere that's using that, but it should be possible.

I've moved the properties back into `popupProps` inside the `PopupTrigger`. At least now it's consistent and clear what those particular properties will be used for...